### PR TITLE
Feature/fui 436 table header sorting

### PR DESCRIPTION
**Pull request by ianhaycox**
Added functionality to sort list tables when clicking on the header.

1st commit is just linting.

Because the data from the list is not received from an API but from within the flow, sorting has been implemented as client-side sorting.

Sorting is performed on the temporary object after filtering and before pagination has been applied to ensure the 'global' model is not modified causing any side affects.

Tests supplied.

### CODE REVIEW

File: __tests__/items-container.test.tsx [hunk 1]
```diff
@@ -334,71 +711,244 @@ describe('ItemsContainer component behaviour', () => {
...
+        expect(objectData).toEqual(sortingTestData);
+    });
+
+    test('Verify sort method handles contentType:boolean', () => {
+        componentWrapper = manyWhoMount();
+        const objectData = Object.assign(sortingTestData);
+
+        objectData.sort(componentWrapper.instance().compare('boolean', false));
+        // Only two unique values so the result of sorting three rows is non-deterministic
+        expect(objectData[0].properties[0].contentValue).toEqual('');
```
Comment (TomFullalove) [inline __tests__/items-container.test.tsx]: Maybe try having one boolean being True, one being False, and one being empty.
Having no False and 2 True seems like we're missing some edge cases.

File: __tests__/items-container.test.tsx [hunk 2]
```diff
@@ -334,71 +711,244 @@ describe('ItemsContainer component behaviour', () => {
...
+                        contentFormat: '',
+                        contentType: 'ContentString',
+                        contentValue: 'my value',
+                        developerName: 'stuff',
+                        objectData: null,
+                    },
+                ],
+            },
+        ];
+        const objectData = Object.assign(single);
```
Comment (TomFullalove) [inline __tests__/items-container.test.tsx]: I don't think these Object.assigns are needed in the end.
Just an = seems fine.

File: __tests__/items-container.test.tsx (hunk 1 above)
Comment (ianhaycox) [reply __tests__/items-container.test.tsx]: Will do

File: __tests__/items-container.test.tsx (hunk 2 above)
Comment (ianhaycox) [reply __tests__/items-container.test.tsx]: = just copies the reference, Object.assign is correct but as you pointed out, I'm doing it wrong. Will change as discussed to use correctly.
